pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2000.go

```diff
@@ -94,18 +98,32 @@ func (r *Rule2000) Run(ctx context.Context) (rule.RuleResult, error) {
 				deniesIngress = true
 			}
 
+			if !allowsAllIngress && slices.ContainsFunc(networkPolicy.Spec.Ingress, func(ingress networkingv1.NetworkPolicyIngressRule) bool {
+				return len(ingress.From) == 0 && len(ingress.Ports) == 0
+			}) && slices.Contains(networkPolicy.Spec.PolicyTypes, networkingv1.PolicyTypeIngress) {
+				allowsAllIngressTarget = allowsAllIngressTarget.With("kind", "networkPolicy", "name", networkPolicy.Name)
+				allowsAllIngress = true
+			}
+
 			if !deniesEgress && len(networkPolicy.Spec.Egress) == 0 &&
 				slices.Contains(networkPolicy.Spec.PolicyTypes, networkingv1.PolicyTypeEgress) {
 				deniesEgressTarget = deniesEgressTarget.With("kind", "networkPolicy", "name", networkPolicy.Name)
 				deniesEgress = true
 			}
 
-			if deniesIngress && deniesEgress {
+			if !allowsAllEgress && slices.ContainsFunc(networkPolicy.Spec.Egress, func(egress networkingv1.NetworkPolicyEgressRule) bool {
+				return len(egress.To) == 0 && len(egress.Ports) == 0
+			}) && slices.Contains(networkPolicy.Spec.PolicyTypes, networkingv1.PolicyTypeEgress) {
+				allowsAllEgressTarget = allowsAllEgressTarget.With("kind", "networkPolicy", "name", networkPolicy.Name)
+				allowsAllEgress = true
+			}
+
+			if deniesIngress && deniesEgress || allowsAllIngress && allowsAllEgress {
 				break
 			}
```

Comment:
```suggestion
			if allowsAllIngress && allowsAllEgress {
				break
			}
```

Let's break only when traffic is allowed, breaking when the traffic is denied might lead us to miss `NetworkPolicy`s that allow traffic.